x/emissions/types/params.go

```diff
@@ -41,27 +30,80 @@ func DefaultParams() Params {
 	return NewParams()
 }
 
-// ParamSetPairs get the params.ParamSet
-func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{
-		paramtypes.NewParamSetPair(KeyPrefix(ParamMaxBondFactor), &p.MaxBondFactor, validateMaxBondFactor),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamMinBondFactor), &p.MinBondFactor, validateMinBondFactor),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamAvgBlockTime), &p.AvgBlockTime, validateAvgBlockTime),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamTargetBondRatio), &p.TargetBondRatio, validateTargetBondRatio),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamValidatorEmissionPercentage), &p.ValidatorEmissionPercentage, validateValidatorEmissionPercentage),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamObserverEmissionPercentage), &p.ObserverEmissionPercentage, validateObserverEmissionPercentage),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamTssSignerEmissionPercentage), &p.TssSignerEmissionPercentage, validateTssEmissionPercentage),
-		paramtypes.NewParamSetPair(KeyPrefix(ParamDurationFactorConstant), &p.DurationFactorConstant, validateDurationFactorConstant),
+// Validate validates the set of params
+func (p Params) Validate() error {
+	err := validateMaxBondFactor(p.MaxBondFactor)
+	if err != nil {
+		return err
+	}
+	err = validateMinBondFactor(p.MinBondFactor)
+	if err != nil {
+		return err
+	}
+	err = validateAvgBlockTime(p.AvgBlockTime)
+	if err != nil {
+		return err
+	}
+	err = validateTargetBondRatio(p.TargetBondRatio)
+	if err != nil {
+		return err
+	}
+	err = validateValidatorEmissionPercentage(p.ValidatorEmissionPercentage)
+	if err != nil {
+		return err
+	}
+	err = validateObserverEmissionPercentage(p.ObserverEmissionPercentage)
+	if err != nil {
+		return err
+	}
+	err = validateTssEmissionPercentage(p.TssSignerEmissionPercentage)
+	if err != nil {
+		return err
+	}
+	err = validateBallotMaturityBlocks(p.BallotMaturityBlocks)
+	if err != nil {
+		return err
+	}
+	return validateObserverSlashAmount(p.ObserverSlashAmount)
```

Comment:
Is there any value in keeping these interface-based methods? or can we just simply it using just the conditions for the values